docs/en/pid-issuing.rst

```diff
@@ -0,0 +1,511 @@
+
+.. note::
+
+    The Wallet Instance redirect URI is a universal or app link registered with the local operating system, so this latter will resolve it and pass the response to the Wallet Instance.
+ 
+.. code-block:: 
+
+    HTTP/1.1 302 Found
+    Location: eudiw://start.wallet.it?code=SplxlOBeZQQYbYS6WxSbIA&state=fyZiOL9Lf2CeKuNT2JzxiLRDink0uPcd
+ 
+**Steps 12-13:** The Wallet Instance creates a key for DPoP and a fresh DPoP proof for the token request to the PID Provider. DPoP provides a way to bind the access token to a certain sender (Wallet Instance) `[DPoP-draft16] <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-16>`_. Thus, it mitigates the misuse of leaked or stolen access tokens at the credential endpoint of PID Provider as the attacker needs to present a valid DPoP proof.
+
+**Step 14:** The Wallet Instance sends a token request to the PID Provider's token endpoint using the authorization *code*, *code_verifier* and *DPoP proof*, and *Wallet Instance Attestation*. The *Wallet Instance Attestation* is used to perform the Wallet Instance authentication at the token endpoint as it is defined in [:rfc:`7521`]. 
+
+.. code-block::
+
+    POST /token HTTP/1.1
+    Host: pid.it
+    Content-Type: application/x-www-form-urlencoded
+    DPoP: eyJ0eXAiOiJkcG9wK2p3dCIsImFsZyI6IkVTMjU2IiwiandrIjp7Imt0eSI6Ik
+        VDIiwieCI6Imw4dEZyaHgtMzR0VjNoUklDUkRZOXpDa0RscEJoRjQyVVFVZldWQVdCR
+        nMiLCJ5IjoiOVZFNGpmX09rX282NHpiVFRsY3VOSmFqSG10NnY5VERWclUwQ2R2R1JE
+        QSIsImNydiI6IlAtMjU2In19.eyJqdGkiOiItQndDM0VTYzZhY2MybFRjIiwiaHRtIj
+        oiUE9TVCIsImh0dSI6Imh0dHBzOi8vc2VydmVyLmV4YW1wbGUuY29tL3Rva2VuIiwia
+        WF0IjoxNTYyMjYyNjE2fQ.2-GxA6T8lP4vfrg8v-FdWP0A0zdrj8igiMLvqRMUvwnQg
+        4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg
+
+    client_id=$thumprint-of-the-jwk-in-the-cnf-wallet-attestation$
+    &grant_type=authorization_code
+    &code=SplxlOBeZQQYbYS6WxSbIA
+    &redirect_uri=eudiw://start.wallet.it
+    &code_verifier=dBjftJeZ4CVP-mB92K27uhbUJU1p1r_wW1gFWFOEjXk
+    &client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer
+    &client_assertion=$WalletInstanceAttestation$
+
+
+**Step 15:** The PID Provider validates the request and if it is successful, it issues an *access token* (bound to the DPoP key) and a fresh *c_nonce*.  
+
+.. code-block::
+
+    HTTP/1.1 200 OK
+    Content-Type: application/json
+    Cache-Control: no-store
+    
+    {
+    "access_token": "Kz~8mXK1EalYznwH-LC-1fBAo.4Ljp~zsPE_NeO.gxU",
+    "token_type": "DPoP",
+    "expires_in": 2677,
+    "c_nonce": "tZign[...]snFbp",
+    "c_nonce_expires_in": 86400
+    }
+
+
+**Steps 16-18:** The Wallet Instance creates a fresh key pair the new credential shall be bound to.  Then, it creates a proof of possession with the new key and the *c_nonce* obtained in **Step 15** and it creates a DPoP proof for the request to the PID credential issuance endpoint.  
```

Comment:
The key pair of the wallet instance is created during the creation of the wallet instance attestation, therefore the PID will be linked to the same public key of the wallet instance attestation previously sent.
Otherwise the PID provider cannot be sure that the wallet instance attestation belongs to the instance with which it is communicating

Reply:
This seems to me to be at odds then with the idea of the SDK as the keys must always be generated by the device. I don't understand the added value of generating a new key pair. It will be up to the wallet instance to decide when to generate new keys and therefore new attestations. @peppelinux what do you think?

Reply:
The keys are always generated by the device so I didn't get the first part of your comment. The wallet instance attestation will be checked again at Step 14 before releasing the Token and the fresh key generation is happening at Step 16, so I don't expect in this short period of time something is going to happen. In this way, you can have ephemeral PID and long-lived wallet instance attestation.

Reply:
The PID is not be bound to the same public key of the wallet instance attestation. The purpose of the holder binding is to verify that the holder has the control over the crittographic material the credential is bound to. We don't require to check that the credential was obtained with the same wallet isntance attestation. If you bind the credential to the same public key of the WA, what happen when the WA expires? You must revoke the credential?

Reply:
ok, @fmarino-ipzs  you convinced me! However, this still implies that the public boundend within the PID must always exist, correct?

Reply:
Yes, we already considered the holder binding for the PID.

Reply:
> even if we try to simplify the flow embedding as many stuffs in a single shot the problem of the multiple presentation is always along the road of the real world scenario. So, let's go for the multiple presentations anyway, it costs more but it guarranties more privacy and flexiblity in time

@peppelinux does that mean we won't have presentations using data coming from multiple (Q)EAAs?

Reply:
This implies that if we have more than one (Q)EAA with the holder binding, each (Q)EAA must be presented separately. If we have multiple (Q)EAA whithout the holder binding (or with only one credential with the binding) they could be presented in a single shot.